File: backend/character-sheet-service/src/main/java/com/magicandphoenix/charactersheetservice/controller/dto/CharacterDTO.java
package com.magicandphoenix.charactersheetservice.controller.dto;

import com.magicandphoenix.charactersheetservice.enums.Country;
import com.magicandphoenix.charactersheetservice.enums.Profession;
import com.magicandphoenix.charactersheetservice.enums.RacialLineage;
import com.magicandphoenix.charactersheetservice.model.Character;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.List;

public class CharacterDTO {


    @NotEmpty
    private String name;

    @NotEmpty
    private Long userId;

    @NotEmpty
    private String racialLineage;

    @NotEmpty
    private String country;

    @NotEmpty
    private String profession;

    //Px, can only be giving by the master
    private int px;

    //Attributes

    private int physical;
    private int skill;
    private int mental;
    private int social;

    //The list with the common aptitudes

    private int athletics;
    private int knowledge;
    private int interpretation;
    private int perception;
    private int caution;
    private int conjuration;

    public CharacterDTO() {
    }

    public CharacterDTO(Character character){
        //basic info
        setName(character.getName());
        setUserId(character.getUserId());
        setRacialLineage(character.getRacialLineage().toString());
        setCountry(character.getCountry().toString());
        setProfession(character.getProfession().toString());
        setPx(character.getPx());

        //Attributes
        setPhysical(character.getPhysical());
        setSkill(character.getSkill());
        setMental(character.getMental());
        setSocial(character.getSocial());

        //Basic skills
        setAthletics(character.getAthletics());
        setKnowledge(character.getKnowledge());
        setInterpretation(character.getInterpretation());
        setPerception(character.getPerception());
        setCaution(character.getCaution());
        setConjuration(character.getConjuration());
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getRacialLineage() {
        return racialLineage;
    }

    public void setRacialLineage(String racialLineage) {
        this.racialLineage = racialLineage;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getProfession() {
        return profession;
    }

    public void setProfession(String profession) {
        this.profession = profession;
    }

    public int getPx() {
        return px;
    }

    public void setPx(int px) {
        this.px = px;
    }

    public int getPhysical() {
        return physical;
    }

    public void setPhysical(int physical) {
        this.physical = physical;
    }

    public int getSkill() {
        return skill;
    }

    public void setSkill(int skill) {
        this.skill = skill;
    }

    public int getMental() {
        return mental;
    }

    public void setMental(int mental) {
        this.mental = mental;
    }

    public int getSocial() {
        return social;
    }

    public void setSocial(int social) {
        this.social = social;
    }

    public int getAthletics() {
        return athletics;
    }

    public void setAthletics(int athletics) {
        this.athletics = athletics;
    }

    public int getKnowledge() {
        return knowledge;
    }

    public void setKnowledge(int knowledge) {
        this.knowledge = knowledge;
    }

    public int getInterpretation() {
        return interpretation;
    }

    public void setInterpretation(int interpretation) {
        this.interpretation = interpretation;
    }

    public int getPerception() {
        return perception;
    }

    public void setPerception(int perception) {
        this.perception = perception;
    }

    public int getCaution() {
        return caution;
    }

    public void setCaution(int caution) {
        this.caution = caution;
    }

    public int getConjuration() {
        return conjuration;
    }

    public void setConjuration(int conjuration) {
        this.conjuration = conjuration;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }
}

File: frontend/magic-and-phoenix/src/app/dto/character-dto.ts
import { Game } from "../models/game";
import { User } from "../models/user";

export class CharacterDTO {
    
    public get conjuration(): number {
        return this._conjuration;
    }
    public set conjuration(value: number) {
        this._conjuration = value;
    }
    public get caution(): number {
        return this._caution;
    }
    public set caution(value: number) {
        this._caution = value;
    }
    public get perception(): number {
        return this._perception;
    }
    public set perception(value: number) {
        this._perception = value;
    }
    public get interpretation(): number {
        return this._interpretation;
    }
    public set interpretation(value: number) {
        this._interpretation = value;
    }
    public get knowledge(): number {
        return this._knowledge;
    }
    public set knowledge(value: number) {
        this._knowledge = value;
    }
    public get athletics(): number {
        return this._athletics;
    }
    public set athletics(value: number) {
        this._athletics = value;
    }
    public get social(): number {
        return this._social;
    }
    public set social(value: number) {
        this._social = value;
    }
    public get skill(): number {
        return this._skill;
    }
    public set skill(value: number) {
        this._skill = value;
    }
    public get mental(): number {
        return this._mental;
    }
    public set mental(value: number) {
        this._mental = value;
    }
    public get physical(): number {
        return this._physical;
    }
    public set physical(value: number) {
        this._physical = value;
    }
    public get px(): number {
        return this._px;
    }
    public set px(value: number) {
        this._px = value;
    }
    public get profession(): String {
        return this._profession;
    }
    public set profession(value: String) {
        this._profession = value;
    }
    public get country(): String {
        return this._country;
    }
    public set country(value: String) {
        this._country = value;
    }
    public get racialLineage(): String {
        return this._racialLineage;
    }
    public set racialLineage(value: String) {
        this._racialLineage = value;
    }

    public get name(): string {
        return this._name;
    }
    public set name(value: string) {
        this._name = value;
    }
    public get userId(): string {
        return this._userId;
    }
    public set user(value: string) {
        this._userId = value;
    }

    constructor(

 
        private _gameIdList: Game[],

        //Basic info
        private _userId: string,
        private _name: string,
        private _racialLineage: String,
        private _country: String,
        private _profession: String,

        //Px, can only be giving by the master
        private _px: number,

        //Attributes
        private _physical: number,
        private _skill: number,
        private _mental: number,
        private _social: number,

        //The list with the common aptitudes

        private _athletics: number,
        private _knowledge: number,
        private _interpretation: number,
        private _perception: number,
        private _caution: number,
        private _conjuration: number


    ){

    }
}